test/models/test_gpt.py

```diff
@@ -135,19 +162,163 @@ def mm_decoder(
     )
 
 
-class TestMultimodalTransformerDecoder:
-    def _pos_ids(self, x):
-        b, seq_len, _ = x.shape
-        pos_ids = torch.arange(seq_len, dtype=torch.long, device=x.device)
-        return pos_ids[None, :]  # (b, seq_len)
+@pytest.fixture
+def gpt(d_model, num_tokens, mm_decoder):
+    class DummyTokenizer(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def encode(self, x):
+            pass
+
+        def decode(self, token_ids):
+            pass
+
+        def lookup(self, token_ids):
+            pass
+
+    def _gpt(in_tokenizer=DummyTokenizer(), out_tokenizer=DummyTokenizer()):
+        return MultimodalGPT(
+            d_model=d_model,
+            num_tokens=num_tokens,
+            in_tokenizer=in_tokenizer,
+            out_tokenizer=out_tokenizer,
+            mm_decoder=mm_decoder,
+        ).eval()
+
+    return _gpt
+
 
+def get_pos_ids(x):
+    b, seq_len, _ = x.shape
+    pos_ids = torch.arange(seq_len, dtype=torch.long, device=x.device)
+    return pos_ids[None, :]  # (b, seq_len)
+
+
+class TestMultimodalGPT:
```

Comment:
could also add test for the ValueErrors in encode and decode

Reply:
Added.